### PR TITLE
3.0 release docs update

### LINKED ISSUE #506: Your documentations has lot of errors eg. Activation::exists() methods returns only bool not an activation object.
### Your Environment

- OS and Version: #.#.#
- PHP Version: #.#.#
- Sentinel Version: #.#.#

*The documentation guide you have provided is having lots of errors.*

### PR DESCRIPTION
**Pull request by 8633brown**
closes #506 
re: #508

### CODE REVIEW

File: usage/authentication.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Key          | Required | Type  | Default | Description
 ------------ | -------- | ----- | ------- | ------------------------------------
 $credentials | true     | array | null    | The user credentials.
 $remember    | false    | bool  | false   | Flag to set the remember cookie.
+login        | false    | bool  | true    | Flag to disable automatic login.
```
Comment (brunogaspar) [inline usage/authentication.md]:
```suggestion
$login       | false    | bool  | true    | Flag to disable automatic login.
```

File: usage/checkpoints.md [hunk 2]
```diff
@@ -21,20 +21,36 @@ The `throttle` checkpoint is responsible for validating the login attempts again
...
-##### Sentinel::addCheckpoint($key, $checkpoint)
+##### Sentinel::addCheckpoint()
 
 Add a new checkpoint.
 
+##### Arguments
+
+Key           | Required | Type                                               | Default | Description
+------------- | -------- | -------------------------------------------------- | ------- | -----------------------------------------
+$key          | true     | string                                             | null    | The name of the checkpoint..
```
Comment (brunogaspar) [inline usage/checkpoints.md]: The descriptions here does not seem to match the argument.

File: usage/checkpoints.md [hunk 3]
```diff
@@ -21,20 +21,36 @@ The `throttle` checkpoint is responsible for validating the login attempts again
...
+$checkpoint   | true     | Cartalyst\Sentinel\Checkpoints\CheckpointInterface | null    | The name of the checkpoint..
```
Comment (brunogaspar) [inline usage/checkpoints.md]: Same

File: usage/checkpoints.md [hunk 4]
```diff
@@ -21,20 +21,36 @@ The `throttle` checkpoint is responsible for validating the login attempts again
...
 ```
 
-##### Sentinel::removeCheckpoint($key);
+##### Sentinel::removeCheckpoint();
+
+##### Arguments
+
+Key           | Required | Type                                               | Default | Description
+------------- | -------- | ------- | ------- | -----------------------------------------
+$key          | true     | string  | null    | The name of the checkpoint.
```
Comment (brunogaspar) [inline usage/checkpoints.md]: Same

File: usage/checkpoints.md [hunk 5]
```diff
@@ -70,6 +92,17 @@ Execute a closure that bypasses all checkpoints.
...
 Bypass all checkpoints.
 
+Returns: result of `$callback`.
+
+##### Arguments
+
+Key           | Required | Type                                               | Default | Description
+------------- | -------- | ------- | ------- | -----------------------------------------
+$callback     | true     | closure | null    | Closure to use when bypassing checkpoints.
+checkpoints   | false    | array   | []      | Array of checkpoints to bypass.
```
Comment (brunogaspar) [inline usage/checkpoints.md]:
```suggestion
$checkpoints  | false    | array   | []      | Array of checkpoints to bypass.
```

File: usage/reminder.md [hunk 6]
```diff
@@ -2,35 +2,56 @@
...
 Creates a new reminder record for the user.
 
-Returns the reminder object.
+Returns: `Cartalyst\Sentinel\Reminders\EloquentReminder`
+
+##### Arguments
+
+Key   | Required | Type                                   | Default | Description
+----- | -------- | -------------------------------------- | ------- | -------------------------
+$user | true     | Cartalyst\Sentinel\Users\UserInterface | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/reminder.md [hunk 7]
```diff
@@ -2,35 +2,56 @@
...
 Check if a reminder record exists for the user.
 
-Returns the reminder object or bool.
+Returns: `bool`
+
+##### Arguments
+
+Key   | Required | Type                                   | Default | Description
+----- | -------- | -------------------------------------- | ------- | -------------------------
+$user | true     | Cartalyst\Sentinel\Users\UserInterface | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/reminder.md [hunk 8]
```diff
@@ -2,35 +2,56 @@
...
+$code | false    | string                                 | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/reminder.md [hunk 9]
```diff
@@ -2,35 +2,56 @@
...
 Attempt to complete the password reset for the user using the code passed and the new password.
 
-Returns bool.
+Returns: `bool`.
+
+##### Arguments
+
+Key       | Required | Type                                   | Default | Description
+--------- | -------- | -------------------------------------- | ------- | -------------------------
+$user     | true     | Cartalyst\Sentinel\Users\UserInterface | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/reminder.md [hunk 10]
```diff
@@ -2,35 +2,56 @@
...
+$code     | true     | string                                 | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/reminder.md [hunk 11]
```diff
@@ -2,35 +2,56 @@
...
+$password | true     | string                                 | null    | The user credentials.
```
Comment (brunogaspar) [inline usage/reminder.md]: Same

File: usage/checkpoints.md [hunk 12]
```diff
@@ -60,16 +80,29 @@ Sentinel::disableCheckpoints();
...
 
 Execute a closure that bypasses all checkpoints.
 
 Bypass all checkpoints.
 
+Returns: result of `$callback`.
+
+##### Arguments
+
+Key           | Required | Type                                               | Default | Description
```
Comment (brunogaspar) [inline usage/checkpoints.md]: Just some extra unnecessary spaces

```suggestion
Key           | Required | Type    | Default | Description
```

File: usage/roles.md [hunk 13]
```diff
@@ -107,6 +107,8 @@ $role = Sentinel::findRoleByName('Admin');
 
 Creates a new role model instance.
 
+Returns: `Cartalyst\Sentinel\Role\EloquentRole`.
```
Comment (suwardany) [inline usage/roles.md]:
```suggestion
Returns: `Cartalyst\Sentinel\Roles\EloquentRole`.
```

**Comment by suwardany** [comment]
Thank you very much @8633brown

**Pull request merged**